src/components/ProjectSearch/TextSearch.js

```diff
@@ -9,54 +9,50 @@ const ManagedTree = createFactory(_ManagedTree);
 import _SearchInput from "../shared/SearchInput";
 const SearchInput = createFactory(_SearchInput);
 
-import searchSources from "../../utils/search/project-search";
-
 import "./TextSearch.css";
 
 export default class TextSearch extends Component {
   constructor(props: Props) {
     super(props);
     this.state = {
-      results: [],
-      inputValue: "",
-      selectedIndex: 0,
-      focused: false
+      inputValue: this.props.query || ""
     };
 
     this.inputOnChange = this.inputOnChange.bind(this);
     this.onKeyDown = this.onKeyDown.bind(this);
     this.close = this.close.bind(this);
+    this.selectMatchItem = this.selectMatchItem.bind(this);
   }
 
   close() {
-    this.setState({ inputValue: "", results: [], selectedIndex: 0 });
     this.props.closeActiveSearch();
   }
 
+  componentDidMount() {
+    if (this.state.inputValue == "") {
+      this.props.loadAllSources();
```

Comment:
another option would be to move this to the searchSources action... 

the advantage is that we always do it before we search. often it will be very fast... and then we know we have everything to search